### PR TITLE
fix: clear debugger logs filter using backspace

### LINKED ISSUE #24990: [Bug]: Unable to clear filters in debugger logs section by pressing 'backspace'
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

User is unable to clear logs filter in debugger by pressing backspace. Though it works fine if the user clicks on "clear button" on the right.

### Steps To Reproduce

1. Open Logs debugger.
2. Press backspace until the filter keyword is cleared.
3. Observed Behavior : The log filter keyword shows up again.
4. Expected Behavior : The log filter keyword should get cleared and not show up again.

### Public Sample App

_No response_

### Environment

Production

### Issue video log

https://www.loom.com/share/e8a26f101c8d402aaa49d66f28d039fc

### Version

Cloud

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
Clearing logs filter in debugger resets the filter with the current entity name. This PR fixes this behaviour by persisting the cleared filter. Adds contains minor changes to remove the padding top that is seen on debugger's errors tab in canvas.
>
>
#### PR fixes following issue(s)
Fixes #24990 
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
#### How Has This Been Tested?
- [x] Manual
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Debugger/JSObjects_navigation_spec.ts [hunk 1]
```diff
@@ -53,4 +53,40 @@ describe("excludeForAirgap", "JSObjects", () => {
...
+    );
+    agHelper.TypeText(
+      debuggerHelper.locators._debuggerFilter,
+      "{backspace}{backspace}{backspace}{backspace}{backspace}{backspace}{backspace}{backspace}{backspace}",
+      { delay: 50, parseSpecialCharSeq: true },
+    );
+    agHelper.AssertText(debuggerHelper.locators._debuggerFilter, "val", "");
+    debuggerHelper.DebuggerLogsFilter("JSObject1");
+    debuggerHelper.DebuggerLogsFilter("{backspace}");
+    agHelper.AssertText(debuggerHelper.locators._debuggerFilter, "val", "");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Debugger/JSObjects_navigation_spec.ts:90]: Can we also add validation for click of clear icon? To make sure that also is works.

**Review by Druthi — APPROVED** [review]

**Review by Aishwarya-U-R — APPROVED** [review]
Approving Cypress changes.

**Review by Druthi — APPROVED** [review]

**Pull request merged**